guides/use-microprofile-amqp-connector-with-ssl-openshift.adoc

```diff
@@ -0,0 +1,194 @@
+= Use MicroProfile Reactive Messaging with AMQP Connector with SSL Connection to AMQ 7 on OpenShift
+:summary: Learn how to securely connect to AMQ 7 deployed on Openshift using microprofile reactive messaging application using AMQP connector.
+:includedir: _includes
+include::{includedir}/_attributes.adoc[]
+:prerequisites-time: 20
+
+In this guide, we will learn how to set up MicroProfile Reactive Messaging application with AMQP Connector
+to connect to AMQ 7 deployed on Openshift. Communication will be secured using SSL/TLS.
+
+include::{includedir}/_prerequisites.adoc[]
+* Access to an OpenShift cluster (try the "Self-managed" variant of local development machine https://developers.redhat.com/products/openshift/download[Red Hat OpenShift
+] for free)
+* https://docs.openshift.com/container-platform/{ocp-version}/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* Install Java `keytool` command
+
+// login to openshift cluster
+include::{includedir}/_proc-log-into-openshift-cluster.adoc[]
+
+== Deploy AMQ 7 using operator
+=== Install AMQ 7 operator
+Follow instruction in https://access.redhat.com/documentation/en-us/red_hat_amq_broker/7.11/html-single/deploying_amq_broker_on_openshift/index#proc-br-installing-operator-to-project-from-operatorhub_broker-ocp[Installing the Operator using OperatorHub]
+
+=== Generate self-signed certificates for keystore and truststore
+To establish an SSL/TLS connection we need keystore and truststore. Here for demonstration purposes we prepare a self-signed certificate.
+
+[NOTE]
+--
+You must be logged by `oc` command for broker certificate to have correct domain of Openshift cluster for hostname. This is necessary for hostname verification during TLS handshake where client is verifying brokers hostname in the certificate.
+--
+
+[source, bash ,options="nowrap"]
+----
+# Generate broker's key pair using obtained OpenShift cluster domain for `apps`
+keytool -genkeypair -alias broker -keyalg RSA -storetype PKCS12 -keystore broker.ks -storepass changeit -validity 365 -dname "CN=*.apps.`oc whoami --show-server=true | sed 's~https://api\.~~;s~:[0-9]*~~'`, OU=My Org Unit, O=My Organization, L=My City, S=My State, C=My Country"
```

Comment:
the `dname` is set differently from the examples in the wildlfy-extras/guides repo?

Reply:
fixed